deploy/test-profile-stack.py

```
@@ -0,0 +1,177 @@
+        self.assertEqual(stack.component_name(["containerd"], "k3s", True,
+                                             "/var/lib/rancher/k3s/data/hash/bin/containerd"), "containerd")
+        self.assertIsNone(stack.component_name(["kubelet"], "notk8s"))
+        self.assertEqual(stack.component_name(["containerd "], "k3s", True,
+            "/usr/local/bin/k3s", "0::/system.slice/k3s.service\n"), "containerd")
+        self.assertIsNone(stack.component_name(["containerd"], "k3s", True,
+            "/usr/bin/containerd", "0::/system.slice/containerd.service\n"))
+        self.assertIsNone(stack.component_name(["containerd-shim-runc-v2"], "k3s", True,
+            "/var/lib/rancher/k3s/data/hash/bin/containerd-shim-runc-v2", "0::/system.slice/k3s.service\n"))
+
+    def test_embedded_component_comparison_is_rejected_before_setup(self):
+        result = subprocess.run(["python3", str(ROOT / "deploy/profile-stack.py"),
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)